src/meta/meta_turtle.c

```diff
@@ -81,12 +93,96 @@ __metadata_load_hot_backup(WT_SESSION_IMPL *session)
         WT_ERR(__wt_getline(session, fs, value));
         if (value->size == 0)
             WT_ERR_PANIC(session, EINVAL, "%s: zero-length value", WT_METADATA_BACKUP);
+        /*
+         * When performing partial backup restore, check through the file metadata entries and check
+         * if the file exists in the file system. If the file doesn't exist, append the filename to
+         * the partial backup list and the partial backup remove list so we can clean up the
+         * metadata and history store afterwards.
+         */
+        metadata_key = (char *)key->data;
+        if (backup_partial_restore) {
+            if (WT_PREFIX_SKIP(metadata_key, "file:")) {
+                if (WT_SUFFIX_MATCH(metadata_key, ".wti") ||
```

Comment:
I attempted to do this, but realised there is one edge that this would fail but should pass. When we expliclity create WT **file:** table, it won't have **.wt** suffix at all, which makes this fail.